packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -196,10 +202,56 @@ export default class CallDiagnosticMetrics {
   }
 
   /**
-   * TODO: NOT IMPLEMENTED
+   * Return Client Event payload by client error code
+   * @param clientErrorCode
+   * @returns
+   */
+  public getErrorPayloadForClientErrorCode(clientErrorCode: number): ClientEventError {
+    let error: ClientEventError;
+
+    if (clientErrorCode) {
+      const partialParsedError = CLIENT_ERROR_CODE_TO_ERROR_PAYLOAD[clientErrorCode];
+
+      if (partialParsedError) {
+        error = merge(
+          {fatal: true, shownToUser: false, name: 'other', category: 'other'}, // default values
+          partialParsedError
+        );
+
+        return error;
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Generate error payload for Client Event
    * @param rawError
    */
-  generateErrorPayload(error: any) {
+  generateClientEventErrorPayload({
+    rawError,
+    parsedError,
+  }: {
+    rawError?: any;
+    parsedError?: ClientEventError;
+  }) {
+    // sometimes we compute the error on the go
+    if (parsedError) {
+      return parsedError;
+    }
+
+    const errorCode = rawError?.body?.errorCode || rawError?.body?.code;
+    if (errorCode) {
+      const clientErrorCode = SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP[errorCode];
+      if (clientErrorCode) {
+        return this.getErrorPayloadForClientErrorCode(clientErrorCode);
+      }
+
+      // by default, send new locus error
+      return this.getErrorPayloadForClientErrorCode(NEW_LOCUS_ERROR_CLIENT_CODE);
```

Comment:
the default might be different depending on the service, maybe we can check the error if it's within a range for specific service (like locus or wbxappapi) and use the corresponding default

Reply:
added